eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClientGetWithContinuationTokenAsyncCollectionResult.cs

```diff
@@ -45,8 +45,9 @@ public override async IAsyncEnumerable<Page<BinaryData>> AsPages(string continua
                 {
                     yield break;
                 }
+                ListWithContinuationTokenResponse result = (ListWithContinuationTokenResponse)response;
                 List<BinaryData> items = new List<BinaryData>();
-                foreach (var item in ((ListWithContinuationTokenResponse)response).Things)
+                foreach (var item in result.Things)
                 {
                     items.Add(BinaryData.FromObjectAsJson(item));
                 }
```

Comment:
the `nextPage` below we should also use `result` instead,
But I did not figure out how to do it without breaking the header case.

```
nextPage = ((ListWithContinuationTokenResponse)response).NextToken;
```